### PR TITLE
HDDS-7737. [HTTPFSGW] Clean up dependencies

### PR DESCRIPTION
**Pull request by dombizita**
## What changes were proposed in this pull request?

Cleaning up dependencies that were added previously and are not used by the HttpFS module. 

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-7737

## How was this patch tested?

Built it successfully locally, also run the `dependency.sh` and the HttpFS related robot tests.

### CODE REVIEW

**Comment by kerneltime** [comment]
@tanvipenumudy can you please help review this ?

File: hadoop-ozone/httpfsgateway/pom.xml [hunk 1]
```diff
@@ -101,36 +101,71 @@
...
     </dependency>
-    <dependency>
-      <groupId>org.apache.curator</groupId>
-      <artifactId>apache-curator</artifactId>
-      <version>2.4.0</version>
-      <type>pom</type>
-    </dependency>
     <dependency>
       <groupId>org.apache.curator</groupId>
       <artifactId>curator-test</artifactId>
```
Comment (adoroszlai) [inline hadoop-ozone/httpfsgateway/pom.xml:111]: It seems `curator-test` is unnecessary:
https://github.com/adoroszlai/hadoop-ozone/actions/runs/3996117051

(Some of) the exclusions need to be moved to `curator-framework` then.

(And don't forget to remove it from `jar-report.txt`, too.)

File: hadoop-ozone/httpfsgateway/pom.xml [hunk 2]
```diff
@@ -101,36 +101,71 @@
...
+        <exclusion>
+          <groupId>org.apache.commons</groupId>
+          <artifactId>commons-math</artifactId>
+        </exclusion>
+      </exclusions>
     </dependency>
     <dependency>
       <groupId>org.apache.curator</groupId>
       <artifactId>curator-framework</artifactId>
       <version>2.4.0</version>
```
Comment (adoroszlai) [inline hadoop-ozone/httpfsgateway/pom.xml:158]: We can add `<scope>runtime</scope>` for `curator-framework`.

File: hadoop-ozone/httpfsgateway/pom.xml [hunk 3]
```diff
@@ -101,36 +101,71 @@
...
+        <exclusion>
+          <groupId>org.slf4j</groupId>
+          <artifactId>slf4j-api</artifactId>
+        </exclusion>
+        <exclusion>
+          <groupId>org.apache.commons</groupId>
+          <artifactId>commons-math</artifactId>
+        </exclusion>
+      </exclusions>
     </dependency>
```
Comment (adoroszlai) [inline hadoop-ozone/httpfsgateway/pom.xml:154]: And we can remove this and let `curator-client` be a transitive dependency via `curator-framework`.

**Review by adoroszlai — CHANGES_REQUESTED** [review]
Thanks @dombizita for the patch.

File: hadoop-ozone/httpfsgateway/pom.xml [hunk 4]
```diff
@@ -101,36 +101,71 @@
       <artifactId>reload4j</artifactId>
       <scope>compile</scope>
     </dependency>
-    <dependency>
-      <groupId>org.slf4j</groupId>
-      <artifactId>slf4j-api</artifactId>
-      <scope>compile</scope>
-    </dependency>
```
[the remark this answers is not in the document]
Comment (dombizita) [reply hadoop-ozone/httpfsgateway/pom.xml]: thanks for bringing this to my attention, I added it back

File: hadoop-ozone/httpfsgateway/pom.xml (hunk 1 above)
Comment (dombizita) [reply hadoop-ozone/httpfsgateway/pom.xml]: thanks for checking it, I removed it from both places.

File: hadoop-ozone/httpfsgateway/pom.xml (hunk 2 above)
Comment (dombizita) [reply hadoop-ozone/httpfsgateway/pom.xml]: thanks, done the changes

File: hadoop-ozone/httpfsgateway/pom.xml (hunk 3 above)
Comment (dombizita) [reply hadoop-ozone/httpfsgateway/pom.xml]: thanks, done changes

**Comment by dombizita** [comment]
thanks for the review and checking these @adoroszlai, I updated my patch accordingly

**Review by adoroszlai — APPROVED** [review]
Thanks @dombizita for updating the patch.

File: hadoop-ozone/httpfsgateway/pom.xml [hunk 5]
```diff
@@ -111,26 +111,37 @@
...
-      <groupId>org.apache.curator</groupId>
-      <artifactId>curator-client</artifactId>
-      <version>2.4.0</version>
-    </dependency>
     <dependency>
       <groupId>org.apache.curator</groupId>
       <artifactId>curator-framework</artifactId>
+      <scope>runtime</scope>
       <version>2.4.0</version>
+      <exclusions>
```
Comment (kerneltime) [inline hadoop-ozone/httpfsgateway/pom.xml:119]: Can you add a comment as to why these dependencies were excluded.

Comment (dombizita) [reply hadoop-ozone/httpfsgateway/pom.xml:121]: thanks for the review @kerneltime, I updated my patch with a comment, that I excluded these as non of them is used by the HttpFS module, but all of them would be a new (unnecessary) dependency in Ozone, the `jar-report.txt` should be updated with that too. I didn't add a comment for each, but I will detail it here.

- the `commons-math` (we are using `commons-math3` in Ozone) and `jline` is never used in the HttpFS module, even in `apache/hadoop`'s HttpFS module (`jline` is used in `hadoop-auth` there, but that was removed from here in #2536). 

- `javax.mail` and `org.jboss.netty` is excluded in `apache/hadoop` as well.

- `log4j` and `slf4j-log4j12` is not used in Ozone anymore, @adoroszlai replaced them in #3773, I use `reload4j` like he did. 

I checked both projects dependency trees to verify these informations. let me know if something is not clear :)

**Comment by adoroszlai** [comment]
Thanks @dombizita for the patch, @kerneltime for the review.

**Pull request merged**